Editor/MenuActions/Object/FreezeTransform.cs

```diff
@@ -43,27 +43,28 @@ public override ActionResult DoAction()
             UndoUtility.RecordMeshAndTransformSelection("Freeze Transforms");
 
             var selection = MeshSelection.topInternal;
-            Vector3[][] positions = new Vector3[selection.Count][];
+                
 
             for (int i = 0, c = selection.Count; i < c; i++)
-                positions[i] = selection[i].VerticesInWorldSpace();
-
-            for (int i = 0, c = selection.Count; i < c; i++)
-            {
+            {                
                 ProBuilderMesh pb = selection[i];
+                TransformUtility.UnparentChildren(pb.transform);
+                bool inParentSpace = (pb.transform.parent != null);
+                Vector3[] positions = pb.VerticesInParentSpace();
```

Comment:
Why not use `TransformUtility.TransformVertex()` and `TransformUtility.InverseTransformVertex()`?